packages/router-plugin/src/router.state.ts

```diff
@@ -118,8 +116,13 @@ export class RouterState {
     });
   }
 
-  private navigationEnd(): void {
-    this.routerStateSnapshot = this._serializer.serialize(this.lastResolvedStateSnapshot);
+  private guardsCheckEnd(routerState: RouterStateSnapshot): void {
+    if (this.routerStateSnapshot) {
+      routerState = { ...this.routerStateSnapshot, ...routerState };
```

Comment:
A problem I forsee here is that this code is relying on the fact that the `this._serializer.serialize(routerState)` returns data that is compatible with the `routerState`.
Maybe we should create a separate snapshot of the data (and any other specific properties that we are trying to resolve this for) directly off the `routerState` and then do this same technique here with that data snapshot.